### PR TITLE
[14.0][ADD] shopfloor_checkout_putinpack_restriction

### PR DESCRIPTION
**Pull request by TDu**
Related PR:
https://github.com/OCA/stock-logistics-workflow/pull/1431
Has been merged.

### CODE REVIEW

**Comment by TDu** [comment]
This is a replacement for https://github.com/OCA/wms/pull/665

File: shopfloor/services/checkout.py [hunk 1]
```diff
@@ -94,17 +94,20 @@ def _response_for_manual_selection(self, message=None):
         data = {"pickings": self.data.pickings(pickings)}
         return self._response(next_state="manual_selection", data=data, message=message)
 
+    def _data_response_for_select_package(self, picking, lines):
+        return {
+            "selected_move_lines": self._data_for_move_lines(lines.sorted()),
+            "picking": self.data.picking(picking),
+            "packing_info": self._data_for_packing_info(picking),
+            "no_package_enabled": not self.options.get("checkout__disable_no_package"),
```
Comment (jbaudoux) [inline shopfloor/services/checkout.py:102]: Can we mark this option as deprecated (replaced by shopfloor_checkout_putinpack_restriction) and to drop in future versions?
cc @simahawk

File: shopfloor_checkout_putinpack_restriction/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+{
+    "name": "Shopfloor Checkout Put In Pack Restriction",
+    "summary": "",
+    "version": "14.0.1.0.0",
+    "development_status": "Alpha",
+    "category": "Inventory",
+    "website": "https://github.com/OCA/wms",
+    "author": "Camptocamp, Odoo Community Association (OCA)",
```
Comment (jbaudoux) [inline shopfloor_checkout_putinpack_restriction/__manifest__.py]:
```suggestion
    "author": "Camptocamp, BCIM, Raumschmiede, Odoo Community Association (OCA)",
```

File: shopfloor_checkout_putinpack_restriction/readme/CONTRIBUTORS.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+* Thierry Ducrest <thierry.ducrest@camptocamp.com>
```
Comment (jbaudoux) [inline shopfloor_checkout_putinpack_restriction/readme/CONTRIBUTORS.rst:1]:
```suggestion
* Thierry Ducrest <thierry.ducrest@camptocamp.com>
* Joshua Lauer <joshua.lauer@raumschmiede.de>
```
cc @raumschmiede-joshuaL

**Review by jbaudoux — APPROVED** [review]
LG, not tested

**Comment by TDu** [comment]
Rebased, squashed.

File: shopfloor_checkout_putinpack_restriction/services/checkout.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+class Checkout(Component):
+    _inherit = "shopfloor.checkout"
+
+    def _data_response_for_select_package(self, picking, lines, message=None):
+        res = super()._data_response_for_select_package(picking, lines)
+        if picking.put_in_pack_restriction == "no_package":
+            res["package_allowed"] = False
+        elif picking.put_in_pack_restriction == "with_package":
+            res["no_package_enabled"] = False
```
Comment (sebalix) [inline shopfloor_checkout_putinpack_restriction/services/checkout.py:16]: I would make the rule clear here, without relying on the default value of `shopfloor`.
```python
        if picking.put_in_pack_restriction == "no_package":
            res["package_allowed"] = False
            res["no_package_enabled"] = True
        elif picking.put_in_pack_restriction == "with_package":
            res["package_allowed"] = True
            res["no_package_enabled"] = False
```
And when we'll remove the `no_package_enabled` option we won't need to think too much.

**Review by sebalix — APPROVED** [review]

**Pull request merged**